runtime/near-vm-logic/src/logic.rs

```diff
@@ -637,6 +639,33 @@ impl<'a> VMLogic<'a> {
         Ok(self.context.block_index)
     }
 
+    /// Gets a block given a height (up to 256)
+    ///
+    /// # Cost
+    ///
+    /// `base`
+    // TODO #1903 rename to `block_height`
+    pub fn get_block_by_hash(&mut self, height_number: BlockHeight) -> Result<u64> {
+        self.gas_counter.pay_base(base)?;
+        let chain_store = self.context.chain_store;
+        let tip = chain_store.header_head().expect("handle this properly");
+        if tip.height == height_number {
+            return Ok(1);
+        }
+        let mut previous_block_hash = tip.prev_block_hash;
+        // iterate up to 255 times more
+        for _ in 0..255 {
+            let block_header =
```

Comment:
We also need to check if previous_block_hash isn’t empty (which indicates we’re at genesis block), i.e. `previous_block_hash == Default::default()` an if so break the loop.

Reply:
nice catch